analysis/analysis.py

```diff
@@ -0,0 +1,297 @@
+import argparse
+import csv
+import fileinput
+import json
+import math
+import os
+import random
+import re
+import sys
+from datetime import datetime, timedelta
+
+import numpy as np
+import plotly.figure_factory as ff
+import plotly.offline as py
+from jinja2 import Environment, FileSystemLoader
+from scipy import stats
+
+
+def index_of_first(lst, pred):
+    for i, v in enumerate(lst):
+        if pred(v):
+            return i
+    return None
+
+
+def has_more_specific_task_bracket(task_bracket, task_indices):
+    for other in task_indices:
+        if other == task_bracket:
+            continue
+        if other[0] in range(task_bracket[0], task_bracket[1]):
+            # theoretically other[1] == end should also be checked for inclusiveness
+            # however we expect to have a tree here where this should not happen
+            return True
+    return False
+
+
+def append_subtask(main_task_name, table_rows, csv_rows, subtasks):
+    ids = set(map(lambda t: t[2]["id"], subtasks))
```

Comment:
A set comprehension would be clearer.
```suggestion
    ids = {t[2]['id'] for t in subtasks}
```